### DISCUSSION TITLE
BUILD FAILED (Ubuntu 20.04 using ruby-build 20230717)

### DISCUSSION THREAD

**Discussion by Adesoji1** [rbenv/ruby-build#2234] [Build failures]
### Steps to reproduce
```
ruby-build 3.1.2 ~/local/ruby-3.1.2 on ubuntu 20.0.4
```

### Expected vs. actual behavior
```
To follow progress, use 'tail -f /tmp/ruby-build.20230803173252.76809.log' or pass --verbose
Downloading ruby-3.1.2.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.1/ruby-3.1.2.tar.gz
Installing ruby-3.1.2...
```
### Logs
```
Inspect or clean up the working tree at /tmp/ruby-build.20230803173252.76809.Rjr4LZ
Results logged to /tmp/ruby-build.20230803173252.76809.log

Last 10 log lines:
installing bundled gem cache:       /home/adesoji/local/ruby-3.1.2/lib/ruby/gems/3.1.0/cache
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Try running `apt-get install -y libssl-dev` to fetch missing dependencies.

Configure options used:
  --prefix=/home/adesoji/local/ruby-3.1.2
  --enable-shared
  LDFLAGS=-L/home/adesoji/local/ruby-3.1.2/lib 
  CPPFLAGS=-I/home/adesoji/local/ruby-3.1.2/include
```

I have installed apt-get install -y libssl-dev`  countless number of times and it was succesful. Help!!!
![rubyissue](https://github.com/rbenv/ruby-build/assets/20457308/7acaea90-9333-4990-8c69-676bc8988b32)

**Comment by mislav** [comment]
Hi, not sure why this failed if you've installed `libssl-dev`! If you observe the `Results logged to <file>` line, you will find a full log file. You could paste that log into a https://gist.github.com and share the link here. Thanks

**Comment by Adesoji1** [comment]
@mislav https://gist.github.com/Adesoji1/94742dba251d7c0d9d4ce291872340f0

**Reply by mislav** [reply]
I have inspected the log but I don't find anything out of the ordinary in there.

Specific Ruby extensions such as `openssl` have their own build log which is in a separate file. When ruby-build prints a BUILD FAILED message, it shows the location of the temporary build directory: `Inspect or clean up the working tree at …`

Inside that directory you can find the `ext/openssl` directory. There should be something like a `mkmf.log` file there. You can inspect that file for potential causes of why the `openssl` extension was built differently.

Please make sure to upgrade ruby-build to latest version (ideally by `git pull` inside the ruby-build installation directory) before trying again.
